src/main/java/edu/cornell/kfs/module/purap/batch/dataaccess/impl/JaggaerUploadDaoJdbc.java

```diff
@@ -116,17 +120,48 @@ public List<JaggaerContractAddressUploadDto> findJaggaerContractAddress(JaggaerC
         }
     }
     
-    private CuSqlQuery buildVendorSql(JaggaerContractUploadProcessingMode processingMode, Date processingDate) {
+    @Override
+    public List<VendorDetail> findVendors(JaggaerUploadSuppliersProcessingMode processingMode, Date processingDate) {
+        try {
+            RowMapper<VendorDetail> rowMapper = (resultSet, rowNumber) -> {
+                if (rowNumber % 100 == 0) {
+                    LOG.info("findVendors, processing row number " + rowNumber);
+                }
+                int headerId = resultSet.getInt(FIELD_NAMES.VNDR_HDR_GNRTD_ID);
+                int detailId = resultSet.getInt(FIELD_NAMES.VNDR_DTL_ASND_ID);
+                VendorDetail vendor = cuVendorService.getVendorDetail(headerId, detailId);
+                return vendor;
+            };
+            CuSqlQuery sqlQuery = buildVendorHeaderIdAndDetailIdSql(processingMode.modeCode, processingDate);
+            logSQL(sqlQuery);
+            return queryForValues(sqlQuery, rowMapper, false);
+        } catch (Exception e) {
+            LOG.error("findVendors, had an error finding vendors: ", e);
```

Comment:
The logSQL() function outputs the SQL and the parameters, that along with the the stack trace should give the troubleshooting details needed.  Is there another piece of information you think is needed?

Reply:
Since we will have the whole query with the data used as parameters, we will be able to run that directly against the DB so I agree that is enough info for now.